ntp-proto/src/time_types.rs

```diff
@@ -1,47 +1,50 @@
 use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign};
-use std::time::Duration;
+use std::time::{Duration, Instant};
 
 /// NtpInstant is a monotonically increasing value modelling the uptime of the NTP service
 ///
 /// It is used to validate packets that we send out, and to order internal operations.
-#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord, Default)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord)]
 pub struct NtpInstant {
-    timestamp: NtpTimestamp,
+    instant: Instant,
 }
 
 impl NtpInstant {
-    pub const ZERO: Self = Self {
-        timestamp: NtpTimestamp::ZERO,
-    };
-
-    pub const fn from_ntp_timestamp(timestamp: NtpTimestamp) -> Self {
-        Self { timestamp }
-    }
-
-    #[cfg(any(test, feature = "fuzz"))]
-    pub(crate) const fn from_bits(bits: [u8; 8]) -> Self {
+    pub fn now() -> Self {
         Self {
-            timestamp: NtpTimestamp::from_bits(bits),
+            instant: Instant::now(),
         }
     }
 
-    pub(crate) const fn to_bits(self) -> [u8; 8] {
-        self.timestamp.to_bits()
+    // used to populate the T1 and T3 fields to ensure the server is doing something sensible
```

Comment:
I think this is (or at least should) only ever be used for the T1 timestamp. T3 isn't even ours to control as that is the timestamp the server indicated it sent its packet at.

Reply:
exactly, so we populate it, and then we check that the server has overwritten what we wrote

Reply:
Sorry, but I think I am missing the point here. The server doesn't write our memory right, so I am not getting what you are implying here